internal/config/config.go

```diff
@@ -174,62 +194,277 @@ func IsGroupsValid(groups []GroupConfig) bool {
 }
 
 type GlobalConfig struct {
-	Routing RoutingConfig `yaml:"routing"`
 	Cache   CacheConfig   `yaml:"cache"`
+	Routing RoutingConfig `yaml:"routing"`
 	Port    int           `yaml:"port"`
 }
 
 func (c *GlobalConfig) setDefaults() {
-	c.Routing.setDefaults()
+	c.Routing.setDefaults(nil, false)
 }
 
 type CacheConfig struct {
 	Redis string `yaml:"redis"`
 }
 
+// ErrorsConfig
+// TODO(polsar): Add the minimum number of requests in the detection window required to apply the error rate.
 type ErrorsConfig struct {
 	HTTPCodes    []string `yaml:"httpCodes"`
 	JSONRPCCodes []string `yaml:"jsonRpcCodes"`
 	ErrorStrings []string `yaml:"errorStrings"`
 	Rate         float64  `yaml:"rate"`
 }
 
+func (c *ErrorsConfig) merge(globalConfig *ErrorsConfig) {
+	if globalConfig == nil {
+		return
+	}
+
+	// TODO(polsar): Can we somehow combine these three sections into one to avoid code duplication?
+	c.HTTPCodes = append(c.HTTPCodes, globalConfig.HTTPCodes...)
+	c.HTTPCodes = sortAndRemoveDuplicates(c.HTTPCodes)
+
+	c.JSONRPCCodes = append(c.JSONRPCCodes, globalConfig.JSONRPCCodes...)
+	c.JSONRPCCodes = sortAndRemoveDuplicates(c.JSONRPCCodes)
+
+	c.ErrorStrings = append(c.ErrorStrings, globalConfig.ErrorStrings...)
+	c.ErrorStrings = sortAndRemoveDuplicates(c.ErrorStrings)
+}
+
+func (c *ErrorsConfig) initialize(globalConfig *RoutingConfig) {
+	var globalErrorsConfig *ErrorsConfig
+	if globalConfig != nil {
+		globalErrorsConfig = globalConfig.Errors
+	}
+
+	if c.Rate == 0 {
+		if globalErrorsConfig == nil {
+			c.Rate = DefaultErrorRate
+		} else {
+			c.Rate = globalErrorsConfig.Rate
+		}
+	}
+}
+
+// Sorts in-place and removes duplicates from the specified slice.
+func sortAndRemoveDuplicates(s []string) []string {
+	slices.Sort(s)
+	return slices.Compact(s)
+}
+
 type MethodConfig struct {
 	Name      string        `yaml:"method"`
 	Threshold time.Duration `yaml:"threshold"`
 }
 
+func (c *MethodConfig) isMethodConfigValid(passiveLatencyChecking bool) bool {
+	if c == nil {
+		return true
+	}
+
+	if passiveLatencyChecking {
+		// TODO(polsar): Validate the method name: https://ethereum.org/en/developers/docs/apis/json-rpc/
+		return !strings.EqualFold(c.Name, LatencyCheckMethod)
+	}
+
+	return true
+}
+
+// LatencyConfig
+// TODO(polsar): Add the minimum number of latencies in the detection window required to apply the threshold.
+// TODO(polsar): Add other aggregation options. Currently, the average of latencies in the detection windows is used.
 type LatencyConfig struct {
+	MethodLatencyThresholds map[string]time.Duration
```

Comment:
Do we need this in addition to the Methods and Threshold fields? Should we keep one or the other?

Reply:
Good idea on embeding within the new struct. 

Let's add some in-line comments about why this field exists here and plans to address